Add option to ignore pending transaction

bhipple opened the pull request:
When pending transactions become final, they may have their amount, transaction
id, date, etc. re-stated. Some accounting import tools are not smart enough to
avoid duplicate entries that have to be manually pruned later.

This commit adds an option for users who are willing to sacrifice some
up-to-date data in order to avoid this problem entirely, by only looking at
final transactions.

bhipple commented:
Tested by importing quite a few transactions and verifying only the pending ones were skipped, which looks like this:
```
[2018/06/24 12:49:18][INFO] reading credentials from plaid-credentials.json
[2018/06/24 12:49:19][INFO] skipping pending transaction for [2018-06-23: Starbucks]
[2018/06/24 12:49:19][INFO] writing record for [2018-06-20: Chipotle Mexican Grill]
...
```

ebridges commented:
Hi -- as a result of configuring deploy to PyPI I've introduced some conflicting changes with this PR.  Would you mind updating your branch so that I can merge?  Thank you.

bhipple commented:
Thanks for all the cleanups and deploying to pypi!  Looks nice!

It took me a while to get to this, but I've now rebased, fixed the merged conflicts, and re-tested with my ignore-pending feature.

ebridges commented:
Awesome! Thanks so much for contributing, it's much appreciated!  :-)

ebridges commented:
Released in [v1.3.1](https://github.com/ebridges/plaid2qif/releases/tag/v1.3.1) also uploaded to PyPi.  Thanks again!
